src/libtest/lib.rs

```diff
@@ -1365,6 +1376,14 @@ pub fn filter_tests(opts: &TestOpts, tests: Vec<TestDescAndFn>) -> Vec<TestDescA
     // Skip tests that match any of the skip filters
     filtered.retain(|test| !opts.skip.iter().any(|sf| matches_filter(test, sf)));
 
+    // Set #[should_panic] tests to ignore
+    if opts.exclude_should_panic {
+        filtered
+            .iter_mut()
+            .filter(|test| test.desc.should_panic != ShouldPanic::No)
+            .for_each(|test| test.desc.ignore = true);
```

Comment:
Alternatively, we could do

```rust
if opts.exclude_should_panic {
    filtered.retain(|test| test.desc.should_panic == ShouldPanic::No);
}
```
which might be more accurate and doesn't mix the flag with ignore tests.

Reply:
That seems cleaner, yeah.